### PR TITLE
(feat) O3-3241 : User Onboarding for Patient Registration 

### PR DESCRIPTION
**Pull request by Vijaykv5**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

 Added [React Joyride library](https://docs.react-joyride.com/) into patient registration module

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-management/assets/94985341/98e372d1-43ba-46c3-b792-4439093a7610

## Related Issue

[O3-3241](https://openmrs.atlassian.net/browse/O3-3241
)

cc : @jayasanka-sack @Piumal1999 

[O3-3241]: https://openmrs.atlassian.net/browse/O3-3241?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Vijaykv5** [comment]
This `Onboarding section` actually renders three times inorder to move into another section. 
so when the User clicks on next for the first time, it renders many times the same section.
@Piumal1999  @jayasanka-sack  @denniskigen Any thoughts on how can we fix up that?

File: packages/esm-patient-registration-app/src/patient-registration/section/section-wrapper.component.tsx [hunk 1]
```diff
@@ -31,6 +39,29 @@ export const SectionWrapper = ({ sectionDefinition, index }: SectionWrapperProps
         {t('allFieldsRequiredText', 'All fields are required unless marked optional')}
       </span>
       <div style={{ margin: '1rem 0 1rem' }}>
+        <Joyride
```
Comment (Piumal1999) [inline packages/esm-patient-registration-app/src/patient-registration/section/section-wrapper.component.tsx:42]: Hi @Vijaykv5, Can we implement this part inside `patient-registration.component.tsx` (or maybe in `root.component.tsx`)? The current wrapper component you used is only a section wrapper, not a wrapper for the entire component.

Comment (Vijaykv5) [reply packages/esm-patient-registration-app/src/patient-registration/section/section-wrapper.component.tsx:42]: Thanks @Piumal1999 Now the rendering get's fixed

Comment (Vijaykv5) [reply packages/esm-patient-registration-app/src/patient-registration/section/section-wrapper.component.tsx]: Thanks @Piumal1999 
The rendering issue got fixed

**Comment by Vijaykv5** [comment]
I've added the UI configuration to enable/disable onboardings


https://github.com/openmrs/openmrs-esm-patient-management/assets/94985341/ec8e0eb1-85ed-45ce-affb-e6408d27725e

**Comment by Piumal1999** [comment]
@Vijaykv5 does it load all 3 steps at the same time? or does it load one after the other? 
Just asking because in the screen recording, both step 2 and 3 are visible at the same time.

**Comment by Vijaykv5** [comment]
> @Vijaykv5 does it load all 3 steps at the same time? or does it load one after the other? Just asking because in the screen recording, both step 2 and 3 are visible at the same time.

@Piumal1999 It will load one after the other.
Sorry the video attached in the description was when the `Joyride` was implemented in the `section-wrapper` component.
<details><summary>Updated Video</summary>
<p>



https://github.com/openmrs/openmrs-esm-patient-management/assets/94985341/e406aea4-913b-457a-93c5-1f2eff5a8774





</p>
</details>

**Comment by ojwanganto** [comment]
Thanks, @Vijaykv5. Implementations will find this feature plus many more to come very useful. It takes a lot of resources to take users through system features. An approach like this one will greatly be a game changer.

**Comment by Vijaykv5** [comment]
Moving this PR to #1181 due to conflicts in `yarn.lock` file.

cc : @jayasanka-sack @Piumal1999

**Pull request closed without merging**